src/generator-typemanager.cs

```diff
@@ -157,7 +157,7 @@ public Type GetUnderlyingNullableType (Type type)
 			return null;
 
 		var gt = type.GetGenericTypeDefinition ();
-		if (gt.Assembly != CorlibAssembly)
+		if (gt.IsNested)
```

Comment:
I agree there's no need to check for the assembly, but is the nested check needed ? or could we just skip it ?

Reply:
It's probably not needed. Then again it's a very cheap check (performance wise), and it rules out a remote possibility if someone does something unwise.